src/test/java/com/jcabi/dynamo/mock/H2DataTest.java

```diff
@@ -111,4 +113,19 @@ public void createsManyTables() throws Exception {
             .with("secondtable", new String[]{"secondid"}, new String[0]);
     }
 
+    /**
+     * H2Data can create tables with long names (max length of DynamoDb table
+     * name is 255 characters).
+     * @throws Exception In case test fails
+     */
+    @Test
```

Comment:
I think it's not enough to test the feature you introduced. If you remove the encoding of table names, this test will also pass. Try it. The issue is mainly with names that contains special chars, like dashes and dots. I would suggest to add another test method which will test table names `"a.b.c"` and `"a-b-c"`